### DISCUSSION TITLE
[Deployment] Error function_selector = 0x, data = 0x

### DISCUSSION THREAD

**Discussion by macaronchef** [zkSync-Community-Hub/zksync-developers#252] [Deployments] [answered]
### Environment

Testnet

### zkSolc Version

v1.3.18

### zksync-web3 Version

I am using "web3dart_macaron: ^2.5.3" package for flutter

### Hardhat.config.ts

```javascript
-
```

### Deployment Script (WITHOUT PRIVATE KEY)

```javascript
-
```

### Package.json

```json
-
```

### Contract Code

```solidity
https://sepolia.explorer.zksync.io/address/0x3b6040DC9b6e776053249aa74AfD281b1393b79F#contract
```

### Does this work on other EVMs? (If yes, please list at least 1 of them)

Same contract and transaction working on BNB smart chain

### Description of What Your Contract Does

AMM Router contract

### Additional Details

I am trying to make a LP adding transaction and it successfully committed sometimes but sometimes get failed with error code `Error function_selector = 0x, data = 0x`. What is the reason of this problem? ABI or Txn details or something else?

Please check transactions below.
Success Txn:
https://sepolia.explorer.zksync.io/tx/0x581b640dc4aaf76829ecc55a61633c73cd01bdfe567d8f2e8e71cc0834edf9ce

Failed Txn:
https://sepolia.explorer.zksync.io/tx/0x2940d03d69327afbf3956dedf6e6b9892a02dd80d32bcbdaf83765ac26bf276b

**Comment by MexicanAce** [comment, marked as answer]
Hi @macaronchef 👋 

This error is a bit generic, but this usually results from a method not existing when it's expected to be available. To investigate further, we recommend using the `era_test_node` tool to replay the transaction locally and investigate for further details: https://era.zksync.io/docs/tools/testing/era-test-node.html#replay-remote-transactions-locally

**Comment by cryptoliang** [comment]
Hi, i have same error, try to increase the `gasLimit` will solve the problem.

**Reply by gauthking** [reply]
to what ?

**Comment by ternianitha** [comment]
The error code "Error function_selector = 0x, data = 0x" typically indicates that the function selector and data payload of the transaction are empty. This could be due to several reasons:

Incorrect ABI: Ensure that the ABI used to encode the transaction data matches the function signatures of the contract.

Invalid Transaction Data: Double-check that the parameters provided in the transaction data are correct and properly formatted according to the contract's function signatures.

SDK or Web3 Library Issue: There might be a bug or compatibility issue with the SDK or Web3 library you're using. Consider updating to the latest version or switching to a different library to see if the issue persists.

Network Connectivity: Inconsistent network connectivity or congestion could also cause transaction data to be improperly transmitted.

Reviewing these factors and debugging the transaction encoding process could help identify the root cause of the issue.

**Comment by franksofa** [comment]
The error code "Error function_selector = 0x, data = 0x" typically indicates that the transaction was submitted with empty function selector and data. This could be due to various reasons such as incorrect contract ABI, incorrect transaction data encoding, or missing parameters in the transaction. Double-check the contract ABI used for encoding the transaction data, ensure all required parameters are included, and verify that the transaction data is correctly formatted according to the contract's function signature.

**Comment by jonnybutlein** [comment]
The error code "function_selector = 0x, data = 0x" indicates that the transaction failed due to an empty function selector and data payload. This could be caused by a variety of factors such as incorrect ABI encoding, invalid transaction parameters, or a bug in the contract logic. To diagnose the issue, carefully review the transaction parameters, contract ABI, and contract code to ensure they are all correct and consistent. Additionally, consider debugging tools and techniques to pinpoint the exact cause of the failure.

**Comment by antoniostoric** [comment]
Issue: Inconsistent success/failure of LP adding transaction with error code "function_selector = 0x, data = 0x".
 Transaction works sporadically; successful and failed examples provided. Contract and transaction function details need review for potential issues.

**Comment by anggabarber** [comment]
It seems like there is an issue with making an LP adding transaction on zkSync testnet using the AMM Router contract. The transaction sometimes gets failed with error code "Error function_selector = 0x, data = 0x". It is unclear what the exact reason for this issue is, and it could be related to the ABI or transaction details. The same contract and transaction appear to be working on Binance Smart Chain. Without further information or access to the code repository, it is difficult to diagnose the exact issue.
